pkg/metrics/metrics.jsx

```diff
@@ -391,37 +400,56 @@ class CurrentMetrics extends React.Component {
                     const is_user = k.match(/^user.*user@\d+\.service.+/);
                     const label = k.replace(/.*\//, '').replace(/\.service$/, '');
                     // only keep cgroup basenames, and drop redundant .service suffix
-                    merged.push([label, v, is_user, false]);
+                    merged.push([label, v, is_user, false, k]);
                 }
-                // filter out podman containers
+                // filter out podman containers, but only for the logged in
+                // user or root user if the user is privileged. Other users
+                // containers will show up under the user@uid cgroup
                 const matches = k.match(podmanCgroupRe);
                 if (matches && v) {
-                    // truncate to 12 chars like the podman output
-                    const containerid = matches.groups.containerid.substr(0, 12);
-                    const is_user = k.match(/^user.slice/);
-                    merged.push([containerid, v, is_user, true]);
+                    let is_user = false;
+                    let uid = 0;
+                    const containerid = matches.groups.containerid;
+                    const umatches = k.match(useridCgroupRe);
+                    if (umatches) {
+                        is_user = true;
+                        uid = parseInt(umatches.groups.userid);
+                    }
+
+                    if (uid === 0 || this.state.userid == uid) {
+                        merged.push([containerid, v, is_user, true, uid]);
+                    }
                 }
             });
             merged.sort((a, b) => b[1] - a[1]);
             return merged.slice(0, n);
-        }
+        };
+
+        const getCachedPodName = (uid, containerid) => {
+            return this.state.podNameMapping[uid] && this.state.podNameMapping[uid][containerid];
```

Comment:
Can you please drop the `{ return }` wrapper and just directly return the value?